benchmark/buffers/buffer_zero.js

```diff
@@ -3,16 +3,18 @@
 const common = require('../common.js');
 
 const bench = common.createBenchmark(main, {
-  n: [1024]
+  n: [1024],
+  type: ['buffer', 'string']
 });
 
-const zero = Buffer.alloc(0);
+const zeroBuffer = Buffer.alloc(0);
+const zeroString = '';
 
 function main(conf) {
   var n = +conf.n;
   bench.start();
   for (let i = 0; i < n * 1024; i++) {
-    Buffer.from(zero);
+    conf.type === 'buffer' ? Buffer.from(zeroBuffer) : Buffer.from(zeroString);
```

Comment:
You mean the `conf.type === 'buffer’` will be run much times？

> 在 2016年3月23日，下午9:53，Ben Noordhuis <notifications@github.com <mailto:notifications@github.com>> 写道：
> 
> In benchmark/buffers/buffer_zero.js <https://github.com/nodejs/node/pull/4414#discussion_r57161722>:
> 
> >  
> >  function main(conf) {
> >    var n = +conf.n;
> >    bench.start();
> >    for (let i = 0; i < n * 1024; i++) {
> > -    Buffer.from(zero);
> > +    conf.type === 'buffer' ? Buffer.from(zeroBuffer) : Buffer.from(zeroString);
> Sorry if I'm being unclear. What I mean is that the overhead of the check and the multiplication may affect the benchmark results. They probably don't but it would be good to check.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/nodejs/node/pull/4414/files/03579ba47a711923c67215d63385b60cdf473c98#r57161722>